src/components/Comments/Comment.js

```diff
@@ -125,18 +127,26 @@ const Comment = ({
           onPress={openProfile}>
           <View style={styles.profile}>
             {comment.data?.anon_user_info_emoji_name || comment.data?.is_anonymous ? (
-              <View
-                style={[styles.image, {backgroundColor: comment.data?.anon_user_info_color_code}]}>
-                <Text>{comment.data?.anon_user_info_emoji_code}</Text>
-              </View>
+              <ProfilePicture
+                karmaScore={comment.karmaScores}
+                size={25}
+                width={6}
+                withKarma
+                isAnon={true}
+                anonBackgroundColor={comment.data?.anon_user_info_color_code}
+                anonEmojiCode={comment.data?.anon_user_info_emoji_code}
+              />
             ) : (
-              <Image
-                source={
+              <ProfilePicture
+                karmaScore={comment.karmaScores}
+                profilePicPath={
                   photo
-                    ? {uri: removeWhiteSpace(photo)}
+                    ? removeWhiteSpace(photo)
                     : require('../../assets/images/ProfileDefault.png')
                 }
-                style={styles.image}
+                size={25}
+                width={6}
+                withKarma
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `ProfilePicture` component is used twice with almost identical props. To follow the DRY (Don't Repeat Yourself) principle, you could create a function that returns this component with the necessary props. This will make your code cleaner and easier to maintain.
```javascript
const renderProfilePicture = (isAnon, photoPath) => {
  const profilePicProps = {
    karmaScore: comment.karmaScores,
    size: 25,
    width: 6,
    withKarma: true,
    ...(isAnon ? 
      {
        isAnon: true,
        anonBackgroundColor: comment.data?.anon_user_info_color_code,
        anonEmojiCode: comment.data?.anon_user_info_emoji_code
      } : 
      {
        profilePicPath: photoPath ? removeWhiteSpace(photoPath) : require('../../assets/images/ProfileDefault.png')
      }
    )
  };

  return <ProfilePicture {...profilePicProps} />;
};

// Then use it like this:
{comment.data?.anon_user_info_emoji_name || comment.data?.is_anonymous ? (
  renderProfilePicture(true)
) : (
  renderProfilePicture(false, photo)
)}
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
agree to disagree, cara ini bisa jadi lebih complex daripada memanggilnya 2 kali